client/driver/logging/rotator.go

```diff
@@ -0,0 +1,218 @@
+package logging
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var (
+	bufSize = 32
+	buf     = make([]byte, bufSize)
+)
+
+// FileRotator writes bytes to a rotated set of files
+type FileRotator struct {
+	// MaxFiles is the maximum number of rotated files allowed in a path
+	MaxFiles int
+	// FileSize is the size a rotated file is allowed to grow
+	FileSize int64
+
+	// path is the path on the file system where the rotated set of files are
+	// opened
+	path string
+	// baseFileName is the base file name of the rotated files
+	baseFileName string
+
+	// logFileIdx is the current index of the rotated files
+	logFileIdx int
+	// oldestLogFileIdx is the index of the oldest log file in a path
+	oldestLogFileIdx int
+
+	// currentFile is the file that is currently getting written
+	currentFile *os.File
+	// currentWr is the number of bytes written to the current file
+	currentWr int64
+
+	logger  *log.Logger
+	purgeCh chan struct{}
+}
+
+// NewFileRotator returns a new file rotator
+func NewFileRotator(path string, baseFile string, maxFiles int,
+	fileSize int64, logger *log.Logger) (*FileRotator, error) {
+	rotator := &FileRotator{
+		MaxFiles: maxFiles,
+		FileSize: fileSize,
+
+		path:         path,
+		baseFileName: baseFile,
+
+		logger:  logger,
+		purgeCh: make(chan struct{}, 1),
+	}
+	if err := rotator.lastFile(); err != nil {
```

Comment:
@dadgar I was wondering about it, we could use nextFile, but imagine the path already has `redis.stdout.100`, we need to start from there and not from 0. it would just take us time to figure out where we should start from.